_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,669 @@
+
+```bash
+  "connectionStrings": [
+    {
+      "connectionString": "mongodb://jhipster-cosmosdb:XBq5KZ81V8hM63KjCOezi1arq...,
+      "description": "Primary MongoDB Connection String"
+    },
+    ...
+  ]
+}
+```
+
+Create a `.env` file in the `docker-compose` subdirectory. Add the following variables to it, substituting your connection string for the placeholder. Make sure the connection string is enclosed in quotes. This value is referenced by the `docker-compose.yml` file and passed to the `store` service, pointing it to the Cosmos DB MongoDB database.
+
+The `ENCRYPT_KEY` will be used as the key for encrypting sensitive values stored in the Spring Cloud Config and used by the JHipster registry. You can put whatever value you want in there. A UUID works well, but any string value will work. The longer, the better.
+
+`docker-compose/.env`
+
+```.env
+SPRING_DATA_MONGO_URI="<your-connection-string>"
+ENCRYPT_KEY=<your-encryption-key
+```
+
+#### Configure Okta OAuth
+
+Use the Okta CLI to create an OIDC (OpenID Connect) application. This is what you need on the Okta side to use Okta as an authentication provider.
+
+If you already have an Okta account, use `okta login` to log into that account with the CLI. Otherwise, use `okta register` to sign up for a free account. 
+
+Create the OIDC app using the following command using a Bash shell opened to the project root.
+
+```bash
+okta apps create jhipster
+```
+
+You can accept the default values by pressing **enter**. If you want to give the app a different, more descriptive, name, that is fine. Just take note of the name because you will need to find the app in the Okta developer dashboard and update the redirect URIs a little later.
+
+This command created a config file named `.okta.env`. It will look something like the following. It helpfully lists the values you will need in the next step.
+
+```bas
+export SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI="https://dev-123456/oauth2/default"
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID="2989u928u383..."
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET="09328uu098u4..."
+```
+
+Both the Docker Compose project and the Kubernetes project use Spring Cloud Config to centralize configuration. The JHipster registry service makes these config values available to all of the other services in the cluster.
+
+Open `docker-compose/central-server-config/application.yml` and add the following to the end, filling in the `issuer-uri`, `client-id`, and `client-secret` taken from the `.okta.env` file. This is the Spring Cloud Config file for the Docker Compose project.
+
+`docker-compose/central-server-config/application.yml`
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://<your-okta-domain>/oauth2/default
+        registration:
+          oidc:
+            client-id: <client-id>
+            client-secret: <client-secret>
+```
```

Comment:
There's a `docker-compose/central-server-config/one-dark.theme.css` file in this directory. Is that a mistake?

Reply:
Yeah, I'll delete it. Crossing the streams.